library/src/permutation/permutation_solution_instances.cpp

```diff
@@ -0,0 +1,361 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (C) 2023-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "permutation_solution_instances.hpp"
+#include "permutation_solution.hpp"
+
+// Ensure access to
+#include "device/hiptensor_permutation_scale_instances.hpp"
+
+namespace hiptensor
+{
+    PermutationSolutionInstances::PermutationSolutionInstances()
+    {
+        // Register all the solutions exactly once
+        // 2d Permutation
+        registerSolutions(
```

Comment:
We may have to 'divide and conquer' on this file by Rank. CI is cutting off builds that are taking too long. It's building for multiple archs so probably best to split this file.

Reply:
Done, added files for instances by rank

Reply:
This file still needs to be split.

Reply:
Done